packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -540,7 +540,7 @@ class RenderOpacity extends RenderProxyBox {
     assert(opacity >= 0.0 && opacity <= 1.0);
   }
 
-  bool get alwaysNeedsCompositing => _alpha != 0 && _alpha != 255;
+  bool get alwaysNeedsCompositing => child != null && (_alpha != 0 && _alpha != 255);
```

Comment:
you need to update the bits when child changes from null to non-null or vice-versa